### PR TITLE
NET-5186 Add NET_BIND_SERVICE capability to consul-dataplane requirements

### PR DESCRIPTION
**Pull request by nathancoleman**
### Description

As a compliment to https://github.com/hashicorp/consul-dataplane/pull/238 and https://github.com/hashicorp/consul-k8s/pull/2787, this PR updates the docs to list the `NET_BIND_SERVICE` capability as a requirement for consul-dataplane.

### Testing & Reproduction steps

N/A

### Links

Related to:
- https://github.com/hashicorp/consul-dataplane/pull/238
- https://github.com/hashicorp/consul-k8s/pull/2787

### PR Checklist

* [ ] updated test coverage
* [x] external facing docs updated
* [ ] appropriate backport labels added
* [x] not a security concern

### CODE REVIEW

**Comment by hc-github-team-consul-core** [comment]
🤔 This PR has changes in the `website/` directory but does not have a `type/docs-cherrypick` label. If the changes are for the next version, this can be ignored. If they are updates to current docs, attach the label to auto cherrypick to the `stable-website` branch after merging.

**Comment by david-yu** [comment]
GKE Autopilot seems to allow the NET_BIND_SERVICE capability: https://cloud.google.com/kubernetes-engine/docs/concepts/autopilot-security#built-in-security 

I'm not sure about AWS Fargate or OpenShift by default.

File: website/content/docs/connect/dataplane/index.mdx [hunk 1]
```diff
@@ -89,3 +89,4 @@ Consul Dataplane supports the following features:
 Be aware of the following limitations and recommendations for Consul Dataplane:
 
 - Consul Dataplane is not supported on Windows.
+- Consul Dataplane requires the `NET_BIND_SERVICE` capability.
```
[the remark this answers is not in the document]
Comment (nathancoleman) [reply website/content/docs/connect/dataplane/index.mdx:92]: 💯 this is perfect, thanks!

**Comment by nathancoleman** [comment]
Waiting to merge until time to cut releases containing the related changes

**Pull request merged**